docs/source/dev/py_postprocessing.rst

```diff
@@ -0,0 +1,178 @@
+.. _development-pytools:
+
+Python postprocessing tool structure
+====================================
+
+For each plugin there are (or have to be implemented) three different classes.
+
+1. A data reader class responsible for loading the data from the simulation directory
+2. A visualizer class that outputs a matplotlib plot
+3. A jupyter-widget class that exposes the parameters of the matplotlib visualizer to the user
+   via other widgets
+
+Data reader
+~~~~~~~~~~~
+The data readers should reside in the ``lib/python/picongpu/plugins/`` directory.
+
+Each reader class needs to implement the following interface functions:
+
+.. code:: python
+
+   def __init__(self, run_directory)
+
+   def get_data_path(self, your_specific_arguments)
+       # has to return the path to the underlying data file
+       # specified by some specific arguments (depending on the plugin output)
+
+   def get_iterations(self, your_specific_arguments)
+       # return a numpy array with iteration numbers for which data is available
+
+   def get(self, your_specific_arguments)
+       # main function of the reader to actually get the data
+
+Visualizer
+~~~~~~~~~~
+The visualizers should reside in the ``lib/python/picongpu/plugins/plot_mpl/`` directory.
+The module names should end on ``_visualizer.py`` and the class name should only be ``Visualizer``.
+
+There is a base class for visualization found in ``base_visualizer.py`` which already handles the plotting logic. It uses the
```

Comment:
As in latex, just start a newline when starting a new sentence. No limit on line length in prose text to break for.

This will make diffs more readable.